### ISSUE TITLE
Mage simulation crashes 725-01 on completion

### ISSUE THREAD

**Issue by Arcanemagus** [simulationcraft/simc#3816]
Whenever a simulation of a mage completes Simulationcraft 725-01 crashes.

#### Simulationcraft Version

Crashing on the 64 bit builds of:
* 8724f66 (+32bit build)
* 6d273b8
* 2f9857b

Working on 64 bit build of 8e93b67.

#### If Simulationcraft is crashing, please include your dxdiag file

[DxDiag.txt](https://github.com/simulationcraft/simc/files/1068744/DxDiag.txt)

#### simc_gui.simc file or contents

<details><summary>simc_gui.simc</summary>

```ini
### Begin GUI options ###
item_db_source=local/bcpapi/wowhead
target_error=0.05
iterations=0
default_world_lag=0.05
max_time=300
vary_combat_length=0.2
fight_style=Patchwerk
tmi_window_global=6
target_level+=3
target_race=Humanoid
default_skill=0.95
optimal_raid=0
override.bloodlust=1
override.bleeding=1
override.mortal_wounds=1
threads=8
process_priority=Below_normal
statistics_level=2
### End GUI options ###
### Begin simulateText ###
# Paste into a SimC profile to compare your current loadout against a proposed loadout

mage="Arcanemagus - Arcane (Equipped)"
talents=2123131
artifact=4:0:0:0:0:72:3:73:1:74:4:75:4:77:4:78:1:79:4:80:1:81:4:82:3:83:3:84:4:86:1:87:1:290:1:1169:1:1339:1:1371:1:1529:4:1530:1:1531:1:1639:1
race=troll
level=110
role=spell
spec=arcane
position=back
head=cruel_gladiators_silk_cowl,id=144627,bonus_id=3568/1679/1507/3336
neck=netherstone_choker,id=134376,bonus_id=1825/1547/3337
shoulder=ravencourt_formal_mantle,id=139247,enchant_id=5441,bonus_id=3573/43/1542/3336
back=goldscar_pelt,id=133639,bonus_id=3573/1532/3528
chest=night_dreamer_robe,id=139089,bonus_id=3573/1547/3336
shirt=artisan_officers_shirt,id=89195
wrist=vileweave_cuffs,id=146792,bonus_id=3394/1547/3337
hands=silken_bloodscent_gloves,id=133609,bonus_id=3573/1808/41/1547/3337,gem_id=130219
waist=cord_of_the_seacaller,id=134433,bonus_id=3573/1532/3528
legs=mystic_kilt_of_the_rune_master,id=132451,bonus_id=3529/3530/1811
feet=boots_of_tirisgarde,id=139747,bonus_id=3385/3526
finger1=legionnaires_band,id=134367,enchant_id=5423,bonus_id=3394/40/1542/3337
finger2=demonic_birthstone_ring,id=141486,enchant_id=5423,bonus_id=3466/1487/3337
trinket1=unstable_arcanocrystal,id=141482,bonus_id=3466/1472
trinket2=obelisk_of_the_void,id=137433,bonus_id=3573/1537/3336
main_hand=aluneth,id=127857,relic_id=3514:1477:3336/3573:1532:3528/3573:1477:3528,bonus_id=729,gem_id=140813/137380/140810
# Relics: Arcana Crux, Rage of the Tides, Farsight Spiritjewel


### End simulateText ###
desired_targets=1
### Begin overrides ###
# User-specified persistent global and player parameters will be set here.

### End overrides ###
### Begin final options ###
### End final options ###
### END ###
```
</details>

#### Reproduction Steps:

It should be enough to simply run the provided `.simc` file in one of the affected versions.

**Comment by vituscze** [comment]
This happens during HTML report generation. The description of Chrono Shift (id 236298) references itself and simc overflows stack when trying to pretty print it.

**Comment by vituscze** [comment]
Fixed in faf69ef45f991c5ac28f3a319494b017cf0baa33

`tooltip_parser_t::parse` now checks for cycles of length 1. Hopefully descriptions/tooltips don't contain cycles of length 2 or more.

**Comment by Arcanemagus** [comment]
Thanks!
